documentation/Data-View-Filters.md

```diff
@@ -0,0 +1,385 @@
+
+The list of options can be configured with the following `filter_parameters` (they are all optional):
+
+- `searchFields`: an array of column names specifying what to filter by. By default each filter just searches the column for which it's been generated (i.e. itself), so you'll rarely need to specify this. But, like the "Address" example [above](#sorting), you might wish to specify a few different database fields to search by (i.e. `address`, `province`, `country`) (You could have also generated a "filter_data" column that combines all three values into a single text field but, since all these values are text, this wouldn't be needed -- just specify the `searchFields` instead.) *The `searchFields` parameter is the only parameter which is common to *all* filters described here.*
+- `delimiter`: this one is important for handling multiple values in a single field, as alluded to [earlier](#creating-filter-data-with-multiple-values). In the example shown in the image here, the filter options are actually derived from text fields structured like this:
+   ```
+   "Ascorbic acid, Agni casti fructus, Alfuzosin and finasteride",
+   "Arsenic, Paracetamol",
+   "Ascorbic acid, Miracle Whip",
+   ...
+   ```
+  In order to seperate them into a more useful list of individual options, we can specify a `delimiter`, which in this case is a comma: `","`. Just a reminder that it preferable to configure multiple options like this as delimited strings rather than arrays (as partial matching array elements in GraphQL is not really feasible in our current setup).
+- `includeNull`: (boolean, default `false`). Sometimes data view columns may contain `null` values. These *will* show up in the filter list (they appear as (Blank)), but only if `null` happens to appear in the top 10 options results (due to 10-item maximum). And typing anything in the Search field will obviously immediately exclude `null`, so it's possible to never see the `null` option in the options list. But if the `includeNull` parameter is set to `true`, then the options list returned from the server will *always* include the `null` "(Blank)" value.
+- `showFilterList`: (boolean, default `true`) as mentioned above, this Options list filter is the default filter for any `string` data. However, if you wish to provide a free-text search instead (often more useful when every item has a different value), set `showFilterList` to `false` and a ["Text search" filter](#text-search) will be displayed instead.
+
+##### API
+
+The options list filter makes its own request to the server to get the list of available options:
+```
+/data-views/<dataViewCode>/filterList/<column>
```

Comment:
Hmmm just wondering why not using the same pattern for URL i.e:
`/data-views/<dataViewCode>/filter-list/<column>`

Reply:
Not sure what URL you're referring to there -- what you wrote is exactly what I have. 🤷‍♂️

Reply:
I was suggesting to change the logic so the URL shows static text in kebab-case

Reply:
Yeah, in the server endpoints, there are quite a few camelCase ones, so probably best to leave as is for now. As long as the user-facing ones are consistent, I'm happy with it. (I'd like to do a bigger tidy-up, but don't want to risk breaking things for the sake of a little cosmetic improvement, but I agree that consistency is better)